source/common/stats/stats_matcher_impl.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/config/metrics/v2/stats.pb.h"
+#include "envoy/stats/stats_matcher.h"
+
+#include "common/common/matchers.h"
+#include "common/protobuf/protobuf.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Stats {
+
+/**
+ * Supplies a stats matcher.
+ */
+class StatsMatcherImpl : public StatsMatcher {
+public:
+  StatsMatcherImpl(const envoy::config::metrics::v2::StatsConfig& config);
+
+  // Default constructor simply allows everything.
+  StatsMatcherImpl() {}
+
+  /**
+   * Take a metric name and report whether or not it should be disallowed.
+   * @param name std::string& a name of Stats::Metric (Counter, Gauge, Histogram).
+   * @return true if that stat should not disallowed.
+   */
+  bool rejects(const std::string& name) const override;
+
+private:
+  // We want to allow stats through by default.
+  bool default_inclusive_ = true;
```

Comment:
nit: it seems strange to try to reflect the default value of the variable in its name. `is_inclusive_` or `inclusive_matching_` along with a comment about the default value seems clearer.

Reply:
That makes sense -- I probably misnamed this while trying to keep straight its default value for myself. I'll rename this `is_inclusive_`.